### ISSUE TITLE
Setting ivec2 with GL.ProgramUniform2 using Vector2i is silently ignored

### ISSUE THREAD

**Issue by typpos** [opentk/opentk#1240]
### Description

Calling `GL.ProgramUniform2` with a `Vecton2i` silently ignores the request.

### Repro steps

glsl:
`ivec2 v;`

C#:
```
Vector2i v = new Vector2i(1,2);
GL.ProgramUniform2(programId, uniformId, v);
```

### Expected behavior

Shader gets the value.

### Actual behavior

Shader does not get the value. Call is silently ignored.

### Related information 

* There is no overload for `GL.ProgramUniform2(..., Vector2i value)`.
* However, there is an implicit cast from `Vector2i` to `Vector2`
* Therefore, trying to set a glsl ivec2 from a Vector2i fails, because OpenGL gets a request to set a pair of floats to an ivec2.

Workaround: explicitly deconstruct the value:  `GL.ProgramUniform2(..., v.X, v.Y)`.

Generalisation: This issue applies for a number of types.

Suggestion: If implicit casts exist, then ensure appropriate overloads for all affected API points are implemented.

Priority: Low. I am using the workaround, but it would make things cleaner and it took me some time to figure this out in a complicated shader.

* Win 10 Pro 10.0.19041
* NVIDIA GTX 1070
* OpenTK 4.5.0 (nuget)
* .NET 5.0

**Comment by NogginBops** [comment]
Ah, this is a mischievous bug this one.
When the integer to float casts for vectors where implemented not that long ago this much have been a missed case where the type of the gl function that is called is determined by the overload (`glProgramUniform2i` vs `glProgramUniform2f`).

This is harder to fix in OpenTK 4 than we would like as the current maintenance team has poor knowledge on how to modify the current bindings generator (it's pretty complicated with a lot of direct IL writing and stuff).

For OpenTK 5 this issue will be solved as are writing a new binder where we will have proper overloads for all of the math types that we have. The new binder will be significantly easier to modify too as it's generating C# code directly instead of rewriting IL code from function stubs, making adding or modifying these types of fixes much easier.

**Comment by NogginBops** [comment]
This is now fixed in `4.7.3`!

I will keep this issue open so we can track it for OpenTK 5 too.

**Comment by NogginBops** [comment]
This is also fixed in OpenTK 5 with #1374
